Conjur/conventions/ruby-style-guide.md

```diff
@@ -24,10 +23,194 @@ This guide is intended to be a "living document" in the sense that it's expected
 as we refine and hone our Ruby knowledge base.
 
 ## Style Baseline
-- **Guide to Ruby** ([Why's Poignant Guide](http://poignant.guide/)) - The best place to start for
-  reading about the langauge
-- **bbatsov's style guide** ([Style Guide](https://github.com/bbatsov/ruby-style-guide)) - A
-  community-driven collection of best practices for Ruby development
+- Follow this official guidelines of the Ruby language https://docs.ruby-lang.org/en/master/doc/syntax/methods_rdoc.html
+  
+- Methods whose purpose is to produce a side effect (change a file, make a network call, raise an error, etc) should have verb phrase names. 
+
+  ```ruby
+  def call
+    validate_and_decode_token
+    get_jwt_identity_from_request
+    validate_host_has_access_to_webservice
+    validate_origin
+    validate_restrictions
+    audit_success
+    new_token
+  rescue => e
+    audit_failure(e)
+    raise e
+  end
+  ```
+
+  
+
+- Methods that return an object should have the name of the returned object so these functions should not have verb phrase names.
+   In some cases, these rules may conflict when a side-effectful method also returns a value. Here the "verb phrase" rule should win since indicating the side effect is more important.
+
+- For low-level style decisions about formatting, etc, we follow the ruby style. We advice to use rubocop to auto correct offence in the files you change before submitting to PR.
+
+- Avoid "name stutter". That is, don't repeat a name from the module path in a class, and don't repeat a name from the class in a method. For example:
+
+   ```ruby
+   # bad
+   class MyCar
+     def start_car_engine
+       # ...
+     end
+   end
+   
+   # good
+   class MyCar
+     def start_engine
+       # ...
+     end
+   end
+   
+   # bad
+   module Authenticators
+     class OktaAuthenticator
+       #...
+     end
+   end
+   
+   # good
+   module Authenticators
+     class Okta
+       #...
+     end
+   end
+   ```
+
+- In command class we should split methods called in `call` function into 2 parts - one with an explicit name so that the command class 'call' function tells the story, and one with the returned object name for memoization.
```

Comment:
Can you please give some more detail on why the second option is better? It looks like we're just adding an extra method call, which feels like it clouds to story. I always prefer concise but understandable over verbose.

Reply:
Removed this rule following this comment
https://github.com/cyberark/community/pull/94#discussion_r691518622